src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -26,13 +26,14 @@ protected internal JsonConverter()
                 HandleNullOnRead = true;
                 HandleNullOnWrite = true;
             }
-            else
+            else if (UsesDefaultHandleNull)
```

Comment:
@layomia @krwq FYI I also updated the default HandleNull logic to only set a tracking flag and moved the HandleNull* initialization logic here. This is to make sure that the default implementation of `HandleNull` is always idempotent.